### PR TITLE
Release 2.0.0

### LINKED ISSUE #16428: Release v2.0.0
### Prepare the project

Do some housekeeping on GitHub in the [main repo](https://github.com/OrchardCMS/OrchardCore).

- [ ] Assign the milestone of the release's version to this issue.
- [ ] Close remaining issues for the version (including merging corresponding pull requests if suitable) or assign them to the next one.
- [ ] Assign all issues that were closed for an upcoming version (including a wildcard version like "1.0.x") to this version (milestone).

### Prepare the code and documentation

Update the source, so everything looks like on the new version.

- [ ] Create a `release/<version name>` branch (e.g., `release/1.8.0`) from `main` or the previous release branch. This is necessary to target pull requests for the upcoming release.
- [ ] Create an issue branch out of this branch as usual.
- [ ] Check the [release_ci](https://github.com/OrchardCMS/OrchardCore/blob/main/.github/workflows/release_ci.yml) workflow is using the expected .NET version to build the Docker images.
- [ ] Update the `OrchardCore.Commons.props` file with `<VersionSuffix></VersionSuffix>` such that preview build numbers are not injected in packages. Verify the `VersionPrefix` tag matches the released version.
- [ ] Update module versions in `src/OrchardCore/OrchardCore.Abstractions/Modules/Manifest/ManifestConstants.cs`.
- [ ] Create a new milestone.
- [ ] Change the Orchard Core version in the project templates, more specifically in `dotnetcli.host.json` and `template.json` files.
- [ ] Add final updates to the release notes in the documentation. It should include the following, at least:
  - Overview of the release's highlights and goals. What do you want people to remember this release for?
  - Prerequisites. What framework version do you need, anything else to work with Orchard?
  - Upgrade steps, any migration necessary from previous versions, and any breaking changes.
- [ ] Add the release notes documentation page to the documentation site's navigation in `mkdocs.yml` and remove it from `not_in_nav`.
- [ ] Update the documentation to mention the version in all places where the latest version is referenced, for example, but not limited to (do a search for the package version string): [Status in the root README](https://docs.orchardcore.net/en/latest/#status), CLI templates, commands, the [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/guides/decoupled-cms/) guide.

### Test the release

Make sure everything works all right.

- [ ] Make sure that [`OrchardCore.Samples` works](https://github.com/OrchardCMS/OrchardCore.Samples).
- [ ] Test the [guides](https://docs.orchardcore.net/en/latest/guides/) with the NuGet packages from the Cloudsmith feed (branches under `release/` are automatically published too). Test at least the following guides:
  - [Creating a modular ASP.NET Core application](https://docs.orchardcore.net/en/latest/guides/create-modular-application-mvc/)
  - [Creating an Orchard Core CMS website](https://docs.orchardcore.net/en/latest/guides/create-cms-application/)
  - [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/guides/decoupled-cms/)
- [ ] Re-certify Orchard Core for the latest major version of Red Hat Enterprise Linux if a new version has been released (e.g., v10 after v9). Refer to Orchard's [Red Hat Ecosystem Catalog profile](https://catalog.redhat.com/software/applications/detail/223797) for the previously certified version, the [Red Hat Customer Portal](https://access.redhat.com/articles/3078) for the latest version, and [our certification guide](https://docs.orchardcore.net/en/latest/topics/red-hat-ecosystem-catalog-certification/) for the certification steps.

### Prepare and publish Orchard Core Translations

Update everything in the [Translations project](https://github.com/OrchardCMS/OrchardCore.Translations). Only do this once all the code changes are done since localized strings can change until then.

- [ ] Update .po files with [PoExtractor](https://github.com/lukaskabrt/PoExtractor). This will also update [Crowdin](https://crowdin.com/project/orchard-core).
- [ ] Publish the new version on NuGet.
- [ ] Update the `OrchardCore.Translations.All` package reference in the main repo's _./Dependencies.Packages.props_ file to refer to the new NuGet package.

### Publish the release

Do the harder parts of making the release public. This should come after everything above is done.

- [ ] Tag the head of `release/<version name>` with the version. Include `v` in the name, e.g. `v1.8.0`.
- [ ] Merge `release/<version name>` to `main`.
  - You'll need to create a pull request.
  - Merge it as a merge commit, not squash merge.
  - If there are merge conflicts, then create a `release/<version name>-integration` branch and fix them there.
- [ ] Create and publish a release [on GitHub](https://github.com/OrchardCMS/OrchardCore/releases/new).
  - Generate release notes.
  - Add a link to the release notes in the docs (something like `For details on this version see the [release notes in the documentation](link here).`). Note that the docs will only be built once the branch is merged to `main`.
- [ ] Test the [guides](https://docs.orchardcore.net/en/latest/guides/) with the packages now automatically published to NuGet. Test at least the following guides:
  - [Creating a modular ASP.NET Core application](https://docs.orchardcore.net/en/latest/guides/create-modular-application-mvc/)
  - [Creating an Orchard Core CMS website](https://docs.orchardcore.net/en/latest/guides/create-cms-application/)
  - [Creating a new decoupled CMS Website](https://docs.orchardcore.net/en/latest/guides/decoupled-cms/)
- [ ] Update [Try Orchard Core](https://github.com/OrchardCMS/TryOrchardCore).

### Publicize the release

Let the whole world know about our shiny new release. Savor this part! These steps will make the release public so only do them once everything else is ready.

- [ ] Tweet
- [ ] Post to the [Orchard Core LinkedIn group](https://www.linkedin.com/groups/13605669/)
- [ ] Post to the [Orchard Core Facebook page](https://www.facebook.com/OrchardCore/)

### After the release is done

- [ ] Create a new milestone with the next release number and close the old milestone.
- [ ] Create a new release notes documentation file for the next version in the `OrchardCore.Docs` project. (e.g., `/releases/1.8.0.md`). Don't add it to the docs navigation and exclude it from validation under `not_in_nav` with `mkdocs.yml`.
- [ ] Update the `OrchardCore.Commons.props` file with the next release number, and `<VersionSuffix>preview</VersionSuffix>` such that preview builds use the new one.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #16428

### CODE REVIEW

**Comment by sebastienros** [comment]
Triggered functional tests successfully after merging main https://github.com/OrchardCMS/OrchardCore/actions/runs/10747067540

File: src/docs/releases/2.0.0.md [hunk 1]
```diff
@@ -1,6 +1,730 @@
...
+- **Enhanced Performance**: Your sites and applications will run faster, smoother, and more efficiently.
+- **Rock-Solid Stability**: We've squashed numerous bugs, ensuring a more reliable and enjoyable development experience.
+- **Seamless Development**: By sticking to version control best practices, we've ensured **no breaking changes** in any patch or minor release, allowing developers to confidently create modules and libraries that will work across the entire 2.x series.
+
+
+### 🎯 Module and Library Developers, Take Note!
+
+Want to create a module or library for Orchard Core 2.0? We've made it easier than ever to build with confidence. Simply set your dependencies to `[2.0, 3.0)` to ensure full compatibility with future updates, without worrying about breaking changes.
+
+## What's Changed
```
Comment (hishamco) [inline src/docs/releases/2.0.0.md:20]: This could be auto-generated in GitHub release

Comment (MikeAlhayek) [reply src/docs/releases/2.0.0.md:20]: It was :)

**Comment by MikeAlhayek** [comment]
@agriffard can you please share the JSON string for TheBootstrapThemeSettings?

**Comment by gvkries** [comment]
@agriffard Yes, when using `dynamic`s, implicit conversions are not working at the moment. You have to cast to the target type manually.

**Comment by agriffard** [comment]
> @agriffard can you please share the JSON string for TheBootstrapThemeSettings?

"TheBootstrapThemeSettings":{"ContentItemId":"4k1qt8jph1ztxy9t381jeqh9nk","ContentItemVersionId":null,"ContentType":"TheBootstrapThemeSettings","DisplayText":null,"Latest":false,"Published":false,"ModifiedUtc":"2020-05-01T19:23:54.9903382Z","PublishedUtc":null,"CreatedUtc":null,"Owner":null,"Author":"admin","TheBootstrapThemeSettings":{"Head":{"Text":"<link type=\"image/x-icon\" rel=\"shortcut icon\" href=\"/TheBootstrapTheme/favicon.ico\" />"},"Foot":{"Text":null},"Background":{"Text":"dark"},"Navigation":{"Text":"FixedTop"},"StickyFooter":{"Value":false},"Bootswatch":{"Text":null}}}

**Comment by MikeAlhayek** [comment]
@agriffard I release a preview with the last fix. Can you please test it out soon and let me know if you have any more issues? As you know, we are getting ready to release 2.0 today or tomorrow

**Comment by agriffard** [comment]
> @agriffard I release a preview with the last fix. Can you please test it out soon and let me know if you have any more issues? As you know, we are getting ready to release 2.0 today or tomorrow

No more errors with version `2.0.0-preview-18337`

**Comment by sebastienros** [comment]
@agriffard can you run the po extractor to get the latest strings in the translations (there are new adminmenu files). Thanks

**Comment by agriffard** [comment]
> @agriffard can you run the po extractor to get the latest strings in the translations (there are new adminmenu files). Thanks

Done: https://github.com/OrchardCMS/OrchardCore.Translations/commit/851324d1534952d2e95a5467f2a0bd095dfd1d43

**Pull request merged**